File: frontend/src/types/voice-call.ts
export type VoiceCallStatus =
  | "initiated"
  | "ringing"
  | "in_progress"
  | "completed"
  | "failed"
  | "no_answer"
  | "busy"
  | "cancelled"

export type VoiceCall = {
  id: string
  user_id: string | null
  candidate_id: string | null
  candidate_name: string | null
  application_id: string | null
  job_title: string | null
  to_number: string
  from_number: string
  status: VoiceCallStatus
  provider_call_id: string | null
  duration_seconds: number | null
  recording_url: string | null
  transcript: string | null
  interview_score: number | null
  evaluation_summary: string | null
  recommendation: string | null
  screening_questions: string[]
  started_at: string | null
  ended_at: string | null
  error_message: string | null
  meta: Record<string, unknown> | null
  created_at: string
  updated_at: string
}

export type VoiceCallListResponse = {
  items: VoiceCall[]
  total: number
  page: number
  page_size: number
  pages: number
}

File: backend/app/schemas/voice_call.py
"""Voice call / Vapi screening schemas."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import VoiceCallStatus


class VoiceCallResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID | None = None
    candidate_id: uuid.UUID | None = None
    candidate_name: str | None = None
    application_id: uuid.UUID | None = None
    job_title: str | None = None
    to_number: str
    from_number: str
    status: VoiceCallStatus
    provider_call_id: str | None = None
    duration_seconds: int | None = None
    recording_url: str | None = None
    transcript: str | None = None
    interview_score: float | None = None
    evaluation_summary: str | None = None
    recommendation: str | None = None
    screening_questions: list[str] = Field(default_factory=list)
    started_at: datetime | None = None
    ended_at: datetime | None = None
    error_message: str | None = None
    meta: dict[str, Any] | None = None
    created_at: datetime
    updated_at: datetime


class VoiceCallListResponse(BaseModel):
    items: list[VoiceCallResponse]
    total: int
    page: int
    page_size: int
    pages: int


class VoiceCallTriggerRequest(BaseModel):
    application_id: uuid.UUID
